tutorials/javaFxPart5.md

```diff
@@ -0,0 +1,419 @@
+
+Children within `AnchorPane` are able to "anchor" their edges to the `AnchorPane` itself if a constraint is set.
+
+Since we want to automatically resize the `TextField` whenever we resize the window, we need to put an anchor
+constraint on the **left and right edges**. Click on the left and right boxes as shown below, and it will automatically
+set a value based on its current position relative to the AnchorPane. Note that the right side is set to 76 as this is
+the width of the button next to it.
+
+![Update AnchorPane constraints of TextField](images/javafx/TextFieldAnchorPaneConstraints2.png)
+
+### Automatically Anchor Send Button
+Since we want to anchor the button to the bottom right corner of the window, we
+set the anchor values for the **bottom and right** side.
+
+![Update AnchorPane constraints of Send Button](images/javafx/SendButtonAnchorPaneConstraints.png)
+
+### Automatically Anchor ScrollPane
+Since we want to resize the chat both horizontally and vertically, we set the anchor pane constraints to **all** edges.
+
+At the same time we **enable Fit To Width** so that the content within the `ScrollPane` resize as well.
+
+![Update AnchorPane constraints of ScrollPane](images/javafx/ScrollPaneAnchorPaneConstraints.png)
+
+Save `MainWindow.fxml` and you're done! Run and test your app out to see if it works when you resize the window.
+
+## Setting limit to Window size
+
+Let's add a few restrictions to the window size so that the user can't resize the window too small. We can do this by
+setting the minimum height and width.
+
+If you don't want to let the user resize horizontally, you can set the maximum width = minimum width.
+
+Update the `start` method as shown below.
+
+**Main.java**
+```java
+public void start(Stage stage) {
+    try {
+        // current code...
+        stage.setMinHeight(220);
+        stage.setMinWidth(417);
+        // stage.setMaxWidth(417); // Add this if you didn't automatically resize elements
+        fxmlLoader.<MainWindow>getController().setDuke(duke);
+        stage.show();
+    } catch (IOException e) {
+        e.printStackTrace();
+    }
+}
+```
+
+# Customize with JavaFX CSS!
+
+For current and future projects, you might need to customize colors, shapes, borders, and other design elements.
+The most effective way to do this is through CSS, so this will be a mini [**JavaFX CSS**](https://docs.oracle.com/javafx/2/api/javafx/scene/doc-files/cssref.html)
+introductory guide.
+
+## Setting up and linking CSS to FXML
+
+Create the following files in `src/main/resources/css`:
+
+**main.css**
+```css
+.root {
+    duke: rgb(237, 255, 242);       /* Create a looked-up color called "duke" within root. */
+    -fx-background-color: duke;
+}
+
+.text-field {
+    -fx-background-color: #d9ffe2;
+    -fx-font: 20px "Arial";
+}
+
+.button {
+    -fx-background-color: mediumspringgreen;
+    -fx-font: italic bold 16px "Arial";
```

Comment:
Using a fixed-width font helps with the alignments. Compare the two below:

[x] First item
[ ] Second item

vs
```
[x] First item
[ ] Second item
```

Reply:
@damithc the varying font size is applied to the send button only, the checks are unaffected (just needed to make adjustments to number of space when a task is not marked):

![image](https://github.com/se-edu/guides/assets/40513197/a44eed57-2ae6-4d48-bafb-eb7aa19e5d32)